server/base/src/main/java/org/apache/accumulo/server/ServerContext.java

```diff
@@ -81,11 +87,19 @@ public ServerContext(ClientInfo info) {
     this(new ServerInfo(info));
   }
 
+  public ServerContext(URL accumuloSiteLocation) {
+    this(SiteConfiguration.getInstance(accumuloSiteLocation));
```

Comment:
This looks like singleton semantics, but I don't think it is, since it depends on the parameter. Should probably use a more descriptive static method name, like "SiteConfiguration.fromFile(file)" and "SiteConfiguration.fromURL(url)", since there shouldn't be just a single instance of this. Either that, or just use a constructor instead of a static method.